rest/importuserxattrtest/import_test.go

```diff
@@ -302,3 +302,90 @@ func TestUserXattrOnDemandImportWrite(t *testing.T) {
 	require.NoError(t, base.JSONUnmarshal(xattrs[base.SyncXattrName], &syncData))
 	assert.Equal(t, []string{channelName}, syncData.Channels.KeySet())
 }
+
+// TestAutoImportUserXattrNoSyncData:
+//   - Define rest tester with sync function assigning channels defined in user xattr
+//   - Insert doc via SDK with user xattr defined to single channel
+//   - Wait for import feed to pick the doc up and assert the doc is correctly assigned the channel defined in user xattr
+//   - Insert another doc via SDK with user xattr defined with array of channels
+//   - Wait for import feed to pick the doc up and assert the doc is correctly assigned the channels defined in user xattr
+func TestAutoImportUserXattrNoSyncData(t *testing.T) {
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyAll)
+
+	rtConfig := rest.RestTesterConfig{
+		SyncFn: `function (doc, oldDoc, meta) {
+   if (meta.xattrs.channels === undefined) {
+      console.log("no user_xattr_key defined");
+      throw ({
+         forbidden: "Missing required property - metadata channel info"
+      });
+   } else {
+      console.log(meta.xattrs.channels);
+      channel(meta.xattrs.channels);
+   }
+}`,
+		DatabaseConfig: &rest.DatabaseConfig{DbConfig: rest.DbConfig{
+			AutoImport:   true,
+			UserXattrKey: base.StringPtr("channels"),
+		}},
+	}
+
+	rt := rest.NewRestTester(t,
+		&rtConfig)
+	defer rt.Close()
+	const (
+		docKey  = "doc1"
+		docKey2 = "doc2"
+	)
+	ctx := base.TestCtx(t)
+	dataStore := rt.GetSingleDataStore()
+
+	userXattrChan := "chan1"
+	userXattrVal := map[string][]byte{
+		"channels": base.MustJSONMarshal(t, userXattrChan),
+	}
+
+	// Write doc with user xattr defined and assert it correctly imports
+	val := make(map[string]interface{})
+	val["test"] = "doc"
+	_, err := dataStore.WriteWithXattrs(ctx, docKey, 0, 0, base.MustJSONMarshal(t, val), userXattrVal, nil)
+	assert.NoError(t, err)
```

Comment:
```suggestion
	require.NoError(t, err)
```

If this fails, it won't get imported.